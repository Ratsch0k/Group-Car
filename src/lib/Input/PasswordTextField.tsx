import React, {useState} from 'react';
import {TextField, InputAdornment, IconButton} from '@material-ui/core';
import VisibilityIcon from '@material-ui/icons/Visibility';
import VisibilityOffIcon from '@material-ui/icons/VisibilityOff';

type TextFieldProps = import('@material-ui/core').TextFieldProps;

export interface PasswordTextFieldProps {
  value?: string;
  onChange?(arg0: React.ChangeEvent<HTMLInputElement>): void;
  size?: TextFieldProps['size'];
  variant: TextFieldProps['variant'];
  id?: string;
  label?: string;
  error?: boolean;
  name?: string;
  helperText?: string;
  onBlur?(event: any): void;
}

const PasswordTextField: React.FC<PasswordTextFieldProps> =
  (props: PasswordTextFieldProps) => {
    const {
      label = 'Password *',
      size = 'medium',
      ...rest
    } = props;

    /**
   * Logic for showing or hiding the password
   */
    const [showPassword, setShowPassword] = useState<boolean>(false);
    const toggleShowPassword = () =>
      setShowPassword((prev: boolean) => !prev);


    return (
      <TextField
        label={label}
        type={showPassword ? 'text' : 'password'}
        InputProps={{
          endAdornment: (
            <InputAdornment position='end'>
              <IconButton onClick={toggleShowPassword}
                edge='end'
                size={size}>
                {
              showPassword ?
              <VisibilityIcon /> :
              <VisibilityOffIcon />
                }
              </IconButton>
            </InputAdornment>
          ),
        }}
        size={size}
        {...rest as any} />
    );
  };

export default PasswordTextField;
