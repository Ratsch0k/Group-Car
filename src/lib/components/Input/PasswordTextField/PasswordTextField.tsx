import React, {useState} from 'react';
import {InputAdornment, IconButton} from '@material-ui/core';
import VisibilityIcon from '@material-ui/icons/Visibility';
import VisibilityOffIcon from '@material-ui/icons/VisibilityOff';
import FormTextField from '../FormTextField';
import {Formik} from 'lib';

type TextFieldProps = import('@material-ui/core').TextFieldProps;

export interface PasswordTextFieldProps {
  value?: string;
  onChange?(arg0: React.ChangeEvent<HTMLInputElement>): void;
  size?: TextFieldProps['size'];
  variant?: TextFieldProps['variant'];
  id?: string;
  label?: string;
  error?: boolean;
  name?: string;
  helperText?: string;
  fullWidth?: boolean;
  onBlur?: TextFieldProps['onBlur'];
  formik?: Formik;
}

export const PasswordTextField: React.FC<PasswordTextFieldProps> =
  (props: PasswordTextFieldProps) => {
    const {
      label = 'Password *',
      size = 'small',
      ...rest
    } = props;

    /**
   * Logic for showing or hiding the password
   */
    const [showPassword, setShowPassword] = useState<boolean>(false);
    const toggleShowPassword = () =>
      setShowPassword((prev: boolean) => !prev);


    /* eslint-disable @typescript-eslint/no-explicit-any */
    return (
      <FormTextField
        label={label}
        type={showPassword ? 'text' : 'password'}
        data-testid='textfield'
        InputProps={{
          endAdornment: (
            <InputAdornment position='end'>
              <IconButton onClick={toggleShowPassword}
                edge='end'
                size={size}>
                {
                  showPassword ?
                    <VisibilityIcon /> :
                    <VisibilityOffIcon />
                }
              </IconButton>
            </InputAdornment>
          ),
        }}
        {...rest as any} />
    );
  };

export default PasswordTextField;
