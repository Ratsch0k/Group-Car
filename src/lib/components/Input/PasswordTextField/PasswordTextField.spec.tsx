import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react';
import PasswordTextField from './PasswordTextField';
import '@testing-library/jest-dom/extend-expect';

it('renders without crashing', () => {
  render(<PasswordTextField />);
});

it('renders with type password usually', () => {
  const {container} = render(<PasswordTextField/>);

  expect(container).toMatchSnapshot();
  expect(container.getElementsByTagName('input')[0])
      .toHaveAttribute('type', 'password');
});

it('type of input changes after button press', () => {
  const {container} = render(<PasswordTextField/>);

  expect(container).toMatchSnapshot();
  expect(container.getElementsByTagName('input')[0])
      .toHaveAttribute('type', 'password');

  fireEvent.click(screen.getByRole('button'));

  expect(container).toMatchSnapshot();
  expect(container.getElementsByTagName('input')[0])
      .toHaveAttribute('type', 'text');

  fireEvent.click(screen.getByRole('button'));

  expect(container).toMatchSnapshot();
  expect(container.getElementsByTagName('input')[0])
      .toHaveAttribute('type', 'password');
});
