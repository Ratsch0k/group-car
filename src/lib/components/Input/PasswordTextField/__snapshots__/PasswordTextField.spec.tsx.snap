// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with type password usually 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    data-testid="textfield"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
      data-shrink="false"
    >
      Password *
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
        name="test"
        type="password"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiIconButton-edgeEnd"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled-7"
        >
          <span>
            Password *
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;

exports[`type of input changes after button press 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    data-testid="textfield"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
      data-shrink="false"
    >
      Password *
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
        name="test"
        type="password"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiIconButton-edgeEnd"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled-11"
        >
          <span>
            Password *
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;

exports[`type of input changes after button press 2`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    data-testid="textfield"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
      data-shrink="false"
    >
      Password *
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
        name="test"
        type="text"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiIconButton-edgeEnd"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled-11"
        >
          <span>
            Password *
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;

exports[`type of input changes after button press 3`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    data-testid="textfield"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
      data-shrink="false"
    >
      Password *
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
        name="test"
        type="password"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiIconButton-edgeEnd"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled-11"
        >
          <span>
            Password *
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
